### PR TITLE
Clarify implicit inheritance for structs

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: docs/csharp/tutorials/inheritance.md [hunk 1]
```diff
@@ -133,12 +133,12 @@ Because of implicit inheritance, you can call any inherited member from a `Simpl
...
-| Type category | Implicitly inherits from                                                      |
-| ------------- | ----------------------------------------------------------------------------- |
-| class         | <xref:System.Object>                                                          |
-| struct        | <xref:System.ValueType>, <xref:System.Object>                                 |
-| enum          | <xref:System.Enum>, <xref:System.ValueType>, <xref:System.Object>             |
-| delegate      | <xref:System.MulticastDelegate>, <xref:System.Delegate>, <xref:System.Object> |
+| Type category                    | Implicitly inherits from                                                      |
+| -------------------------------- | ----------------------------------------------------------------------------- |
+| class                            | <xref:System.Object>                                                          |
+| struct (excluding `ref` structs) | <xref:System.ValueType>, <xref:System.Object>                                 |
```
Comment (Youssef1313) [inline docs/csharp/tutorials/inheritance.md:139]: Needs confirmation that my change is correct before merging.

Comment (BillWagner) [reply docs/csharp/tutorials/inheritance.md:139]: ping @jaredpar @MadsTorgersen 

Is this how we'd want to explain the implicit inheritance for `ref struct` types?

**Comment by BillWagner** [comment]
closing based on offline conversation with @jaredpar

File: docs/csharp/tutorials/inheritance.md (hunk 1 above)
Comment (Youssef1313) [reply docs/csharp/tutorials/inheritance.md:139]: @BillWagner Thanks. But why the compiler says the opposite?

image

Comment (jaredpar) [reply docs/csharp/tutorials/inheritance.md:139]: Diagnostic clarity issue. I'm a bit surprised we didn't make that an error though.

Comment (Youssef1313) [reply docs/csharp/tutorials/inheritance.md:139]: @jaredpar Thanks. Opened https://github.com/dotnet/roslyn/issues/50609 to improve this scenario.

Comment (BillWagner) [reply docs/csharp/tutorials/inheritance.md:139]: @Youssef1313 Interestingly, if you use the pattern matching `is`, it generates CS8121 (an error):

```csharp
(if s is object o)
```

**Pull request closed without merging**